pkg/blob/controllerserver.go

```diff
@@ -667,6 +683,106 @@ func (d *Driver) DeleteBlobContainer(ctx context.Context, subsID, resourceGroupN
+	return wait.ExponentialBackoff(d.cloud.RequestBackoff(), func() (bool, error) {
+		var err error
+		if len(secrets) > 0 {
+			accountSasToken, rerr := getAccountSasToken(secrets, accountName)
+			if rerr != nil || accountSasToken == "" {
+				klog.V(2).Infof("get account sas token from secrets err: %s, generate sas token for account(%s)", err.Error(), accountName)
+				accountSasToken, rerr = GenerateSASToken(accountName, accountKey, blobUrl)
+				if rerr != nil {
+					return true, rerr
+				}
+			}
+			srcContainer, getErr := getContainerReference(srcContainerName, secrets, d.cloud.Environment)
+			if getErr != nil {
+				return true, getErr
+			}
+			srcExist, _ := srcContainer.Exists()
+			if !srcExist {
+				return true, fmt.Errorf("srcContainer(%s) is not exist", srcContainerName)
+			}
+			srcPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, srcContainerName, accountSasToken)
+			dstPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, dstContainerName, accountSasToken)
+
+			klog.V(2).Infof("copy blob container %s to %s", srcContainerName, dstContainerName)
+			out, err := exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false").CombinedOutput()
+			if err != nil {
+				return true, status.Errorf(codes.Internal, "failed to copy blob container %v: %v", err, string(out))
+			}
+			if waitErr := d.waitForBlobContainerCopy(ctx, subsID, resourceGroupName, accountName, dstContainerName, secrets, waitForBlobContainerCopyInterval, waitForBlobContainerCopyTimeout); waitErr != nil {
+				return true, status.Error(codes.Internal, fmt.Sprintf("failed to wait for blob container copy: %v", waitErr.Error()))
+			}
+			klog.V(2).Infof("copied blob container %s to %s", srcContainerName, dstContainerName)
+		} else {
+			klog.V(2).Infof("generate sas token for account(%s)", accountName)
+			accountSasToken, genErr := GenerateSASToken(accountName, accountKey, blobUrl)
+			if genErr != nil {
+				return true, genErr
+			}
+			_, rerr := d.cloud.BlobClient.GetContainer(ctx, subsID, resourceGroupName, accountName, srcContainerName)
+			if rerr != nil {
+				return true, fmt.Errorf("srcContainer(%s) get error: %s", srcContainerName, rerr.Error())
+			}
+			srcPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, srcContainerName, accountSasToken)
+			dstPath := fmt.Sprintf("'https://%s.%s/%s%s'", accountName, blobUrl, dstContainerName, accountSasToken)
+
+			klog.V(2).Infof("copy blob container %s to %s", srcContainerName, dstContainerName)
+			out, err := exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false").CombinedOutput()
+			if err != nil {
+				return true, status.Errorf(codes.Internal, "failed to copy blob container %v: %v", err, string(out))
+			}
+			if waitErr := d.waitForBlobContainerCopy(ctx, subsID, resourceGroupName, accountName, dstContainerName, secrets, waitForBlobContainerCopyInterval, waitForBlobContainerCopyTimeout); waitErr != nil {
```

Comment:
if L743 is sync copy, then it won't be necessary to waitForBlobContainerCopy

Reply:
ok, I have removed this func